### ISSUE TITLE
Footer seems to cause "Segmentation fault (core dumped)" in Ubuntu but not in Windows

### ISSUE THREAD

**Issue by fredrikljung93** [wkhtmltopdf/wkhtmltopdf#3366]
**WkHtmlToPdf version:** 
0.12.4 (with patched qt)

**Affected operating systems:** 
Ubuntu 16.04.1 LTS (32 bit), Ubuntu 16.04.1 LTS (64-bit)

**_Not affected operating systems:_**
_Windows 10 64 bit_

------------------------------------------------

The following commando runs fine on Windows, but not on Ubuntu:
```
wkhtmltopdf --header-spacing 7 --margin-bottom 15 --header-html headeralt.html --footer-html footer.html maskedbad.html test.pdf
```
In ubuntu 32 bit I get the following output in terminal:

> Loading pages (1/6)
> 
> Counting pages (2/6)                                               
> 
> Resolving links (4/6)                                                       
> 
> Loading headers and footers (5/6)                                           
> 
> Printing pages (6/6)                                               
> 
> Segmentation fault (core dumped)=============================] Page 6 of 5

I have used many similar html's, but this particular one is rendering issues. Since the error continues to exist after masking the actual data I was using, I would assume the issue has to do with the number of table rows.

Also, I find the last message funny - "Page 6 of 5"?

However, this issue does not exist in this particular case when removing the footer, and I have never seen anything like this before adding the footer, but I need the empty footer to get the margin-bottom working.

**EDIT:** It seems like margin-bottom between 7mm and 41mm causes segmentation fault. The program runs successfully when I use smaller or bigger margin-bottom.

**EDIT 2:**

For anyone facing similar issues with WkhtmltoPdf - my workaround is to add linebreaks at end of html body and try again. The workaround solves this particular case at least.

Pseudocode:

```
int maxNumberOfRetries = 3;
int retryCount = 0;

string html = "<html><body>somehtml</body></html>";
byte[] pdf = null;

while(retryCount< maxNumberOfRetries)
{
    var process = GetWkHtmlToPdfProcess();
    process.StandardInput.Write(html);
    pdf = process.StandardOutput.Read();
    
    if(process.ExitCode==0)
    {
        break; // Everything went fine
    }
    else
    {
        html = html.Replace("</body>", "<br><br><br><br><br><br><br><br></body>");
        retryCount++;
    }
}
return pdf;
```

[headeralt.html.txt](https://github.com/wkhtmltopdf/wkhtmltopdf/files/816777/headeralt.html.txt)
[maskedbad.html.txt](https://github.com/wkhtmltopdf/wkhtmltopdf/files/816776/maskedbad.html.txt)
[footer.html.txt](https://github.com/wkhtmltopdf/wkhtmltopdf/files/816775/footer.html.txt)

**Comment by fredrikljung93** [comment]
I just discovered that it runs fine when increasing --margin-bottom to 42mm (41mm is not enough lol)

**Comment by ashkulz** [comment]
This looks like a [known issue with patch](https://github.com/wkhtmltopdf/wkhtmltopdf/issues/2894#issuecomment-215411312), but still not found the root cause.

**Comment by ashkulz** [comment]
This should be fixed with `0.12.5-rc`, see #2711 for details.
